Editor/AGS.Editor/Components/AudioComponent.cs

```diff
@@ -602,6 +602,35 @@ private void CommandForceReimportOfAllAudioClips()
             Factory.GUIController.ShowOutputPanel(errors);
         }
 
+        public void ReplaceAudioClipSource(AudioClip clip)
+        {
+            string dir = Path.GetDirectoryName(clip.SourceFileName);
+            if (!Directory.Exists(dir)) dir = Factory.AGSEditor.CurrentGame.DirectoryPath;
+            string selectedFile = _guiController.ShowOpenFileDialog("Select audio to add", AUDIO_FILES_FILTER, dir);
+            if (string.IsNullOrEmpty(selectedFile) || !File.Exists(selectedFile)) return;
+
+            string fileExtension = Path.GetExtension(selectedFile).ToLower();
+            if (!_fileTypeMappings.ContainsKey(fileExtension)) return;
+
+            selectedFile = Utilities.GetRelativeToProjectPath(selectedFile);
+            clip.FileType = _fileTypeMappings[fileExtension];
+            clip.SourceFileName = selectedFile;
+            Path.ChangeExtension(clip.CacheFileName, fileExtension);
+

```

Comment:
I suggest to pick out a method that assigns a source to the clip, and share the same code from both ReplaceAudioClipSource and CreateAudioClipForFile. This is to ensure that whenever a source file is set, same operations are performed. For example, CreateAudioClipForFile also updates FileLastModifiedDate and game.FilesAddedOrRemoved, but you do not do that here.
Such shared method could also call ForceReimportOfAudioClip (or maybe you could rename that into something more suitable. like "recreate cache file").

Reply:
I think on creation there is some ID stuff being made along and some other things we don't want to change, to reuse code from there I need to think a bit on this.

Reply:
> I think on creation there is some ID stuff being made along and some other things we don't want to change, to reuse code from there I need to think a bit on this.

Yes of course, this refers only to a limited set of actions. Probably starting with this line:
https://github.com/adventuregamestudio/ags/blob/c03e186cdcb3f35c707d53301ff97bff6359d515/Editor/AGS.Editor/Components/AudioComponent.cs#L323

Reply:
I could not make such a big method and went for a smaller scope, due to how cache is generated from the index and not a string one can easily change.